llms/local/localllm.go

```diff
@@ -60,15 +60,27 @@ func (o *LLM) Generate(prompts []string) ([]*llms.Generation, error) {
 // New creates a new local LLM implementation.
 func New() (*LLM, error) {
 	// Require the user to set the path to the local LLM binary
-	bin := os.Getenv(localLLMBinVarName)
-	path, err := exec.LookPath(bin)
-	if err != nil {
-		return nil, errors.Join(ErrMissingBin, err)
-	}
+	binPath := os.Getenv(localLLMBinVarName)
 
 	// Allow the user to pass CLI arguments to the local LLM binary (optional)
 	args := os.Getenv(localLLMArgsVarName)
-	c, err := localclient.New(path, args)
+
+	// Expand tilde in the path
+	if binPath[0] == '~' {
```

Comment:
```suggestion
	if strings.HasPrefix(binPath, "~") {
```

Reply:
can we put this into a helper function?

Reply:
Let me do some research on filepath handling... there might be existing libraries can help us.

Reply:
Hmm, it looks like we have to wrap the helper functions by ourselves.

@rdbell How about creating a new `util` package and place the `~` determine and resolving function? And don't forget to create a relevant `doc.go` file with the package comments in it.

@tmc What do you think?